pkg/machines/components/vmOverviewTabLibvirt.jsx

```diff
@@ -137,43 +171,104 @@ class VmOverviewTabLibvirt extends React.Component {
                         {getBootOrder(vm)}
                     </button>
                     { vm.state === "running" && bootOrderChanged() && <WarningInactive iconId="boot-order-tooltip" tooltipId="tip-boot-order" /> }
                 </div>
             );
 
-            memoryLink = (
-                <button className="link-button" id={`${idPrefix}-memory-count`} onClick={this.openMemory}>
-                    {memoryLink}
-                </button>
+            memoryLink = ( // Garrett pls fix this ugliness here
+                <div>
+                    <button className="link-button" id={`${idPrefix}-memory-count`} onClick={this.openMemory}>
+                        {memoryLink}
+                    </button>
+                </div>
             );
         }
         const vcpuLink = (
-            <>
-                <button className="link-button" id={`${vmId(vm.name)}-vcpus-count`} onClick={this.openVcpu}>{vm.vcpus.count}</button>
+            <div>
+                <button className="link-button" id={`${idPrefix}-vcpus-count`} onClick={this.openVcpu}>{vm.vcpus.count}</button>
                 { vm.state === "running" && vcpusChanged && <WarningInactive iconId="vcpus-tooltip" tooltipId="tip-vcpus" /> }
-            </>
+            </div>
         );
 
-        const items = [
-            { title: commonTitles.MEMORY, value: memoryLink, idPostfix: 'memory' },
-            { title: _("Emulated Machine"), value: vm.emulatedMachine, idPostfix: 'emulatedmachine' },
-            { title: commonTitles.CPUS, value: vcpuLink, idPostfix: 'vcpus' },
-            { title: _("Boot Order"), value: bootOrder, idPostfix: 'bootorder' },
-            { title: _("CPU Type"), value: vm.cpu.model, idPostfix: 'cputype' },
-            { title: _("Autostart"), value: autostart, idPostfix: 'autostart' },
-        ];
+        let firmwareLinkWrapper;
+        if (config.provider.name == 'LibvirtDBus' && this.state.loaderElems) {
+            const hasInstallPhase = vm.metadata.hasInstallPhase;
+            const currentFirmware = vm.firmware || label_for_firmware_path(vm.loader, vm.arch);
+
+            /* If the VM hasn't an install phase then don't show a link, just the text  */
+            if (!config.provider.canInstall(vm.state, hasInstallPhase)) {
+                firmwareLinkWrapper = <div id={`${idPrefix}-firmware`}>{currentFirmware == "efi" ? _("UEFI") : _("BIOS")}</div>;
+            } else {
+                const uefiPaths = this.getOVMFBinariesOnHost(this.state.loaderElems);
+
+                const firmwareLink = disabled => {
+                    return ( // Garrett pls fix this
+                        <div>
+                            <button className="link-button" id={`${idPrefix}-firmware`} disabled={disabled} onClick={this.openFirmware}>
+                                {currentFirmware == "efi" ? _("UEFI") : _("BIOS")}
```

Comment:
Is it possible in qemu/kvm/libvirt to user something else from UEFI or BIOS? By just reading docs or looking at xml definitions, I can see that user can ignore firmware attribute (from libvirts `os` element in domains xml), and then set loader to some other type of boot firmware. I see that if only loader is supplied in xml, virt-manager checks whetever it's eufi or bios match, and if not, they proclaim that firmware is "Custom". There are other boot firmwares like LinuxBoot or coreboot.

Reply:
But I'm not 100% sure, I can try to discuss it with libvirt team tomorrow.